src/Ng/Jobs/Feed2MonitoringJob.cs

```diff
@@ -0,0 +1,190 @@
+
+namespace Ng.Jobs
+{
+    public class Feed2MonitoringJob : LoopingNgJob
+    {
+        private V2FeedParser _feedParser;
+        private IPackageMonitoringStatusService _statusService;
+        private IStorageQueue<PackageValidatorContext> _packageValidatorQueue;
+        private ReadCursor _front;
+
+        // Start at -1 because the first thread will increment it to 0 and then fetch that page.
+        private int _pagesProcessed = -1;
+
+        private const int _degreeOfParallelism = 32;
+        private const int _pageSize = 100;
+        private const int _maxRequestFailures = 5;
+
+        public Feed2MonitoringJob(ILoggerFactory loggerFactory)
+            : base(loggerFactory)
+        {
+        }
+
+        public override string GetUsage()
+        {
+            return "Usage: ng endpointmonitoringvalid "
+                   + $"-{Arguments.Gallery} <v2-feed-address> "
+                   + $"-{Arguments.StorageBaseAddress} <storage-base-address> "
+                   + $"-{Arguments.StorageType} azure "
+                   + $"[-{Arguments.StoragePath} <path>]"
+                   + $"[-{Arguments.PackageStatusFolder} <folder>]"
+                   + $"[-{Arguments.Verbose} true|false] ";
+        }
+
+        protected override void Init(IDictionary<string, string> arguments, CancellationToken cancellationToken)
+        {
+            var gallery = arguments.GetOrThrow<string>(Arguments.Gallery);
+            var verbose = arguments.GetOrDefault(Arguments.Verbose, false);
+
+            if (arguments.GetOrThrow<string>(Arguments.StorageType) != Arguments.AzureStorageType)
+            {
+                throw new ArgumentException("File storage is not supported!");
+            }
+
+            var monitoringStorageFactory = CommandHelpers.CreateStorageFactory(arguments, verbose);
+
+            _statusService = CommandHelpers.GetPackageMonitoringStatusService(arguments, monitoringStorageFactory, LoggerFactory);
+
+            _packageValidatorQueue = CommandHelpers.CreateStorageQueue<PackageValidatorContext>(arguments);
+
+            _feedParser = new V2FeedParser(HttpSource.Create(Repository.Factory.GetCoreV2(new PackageSource(gallery))), gallery);
+
+            _front = ValidationCollectorFactory.GetFront(monitoringStorageFactory);
+        }
+
+        protected override async Task RunInternal(CancellationToken cancellationToken)
+        {
+            var validPackagesSet = await GetValidPackages(cancellationToken);
+
+            var invalidPackages = new ConcurrentQueue<FeedPackageIdentity>();
+            _pagesProcessed = 0;
+
+            await ParallelExecutionUtility.ProcessInParallel(() => ProcessPackages(validPackagesSet, invalidPackages, cancellationToken), _degreeOfParallelism);
+
+            if (!invalidPackages.IsEmpty)
+            {
+                var invalidPackagesList = new List<FeedPackageIdentity>();
+                Logger.LogError("The following packages are missing:");
```

Comment:
>LogError [](start = 23, length = 8)

I think that seeing a bunch of error logs with just ids and versions may be confusing. Maybe change the first `LogError` to "<X> packages are missing." and the second `LogError` message to "{id} {version} is missing."